src/Text.Analyzers/Core (Net4)/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -0,0 +1,416 @@
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule
+        );
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            // Do everything within a CompilationStartAction so we have access to the assembly name before the other actions run
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var assemblyName = compilationStartContext.Compilation.AssemblyName;
+
+                // Store the assembly name each time this analyzer runs so the appropriate custom dictionaries can be retrieved from the cache.
+                // The cache will remain in memory until Visual Studio is closed, so it's possible multiple solutions containing the same
+                // assembly names will be opened while the cache persists. This is not a problem because the cache associating assembly name
+                // to dictionaries is refreshed on the call to LoadDictionaries.
+                _namingService = new NamingService(assemblyName);
+                _namingService.LoadDictionaries(compilationStartContext.Options.AdditionalFiles, compilationStartContext.CancellationToken);
+
+                // CompilationStartAction doesn't support reporting diagnostics, so we have to check the assembly name in CompilationEndAction
+                compilationStartContext.RegisterCompilationEndAction(compilationEndContext =>
+                {
+                    if (assemblyName.Length == 1)
+                    {
+                        var diagnostic = Diagnostic.Create(AssemblyMoreMeaningfulNameRule, Location.None, assemblyName);
+                        compilationEndContext.ReportDiagnostic(diagnostic);
+                    }
+                    else
+                    {
+                        foreach (var misspelledWord in GetMisspelledWords(assemblyName))
+                        {
+                            var diagnostic = Diagnostic.Create(AssemblyRule, Location.None, misspelledWord, assemblyName);
+                            compilationEndContext.ReportDiagnostic(diagnostic);
+                        }
+                    }
+                });
+
+                // Find IdentifierTokens with the syntax tree because SyntaxNodeAction only filters on "methods and above"
+                // Reference: http://stackoverflow.com/a/28004943
```

Comment:
The referenced issue does not find **symbols** for locals and parameters. However, `SyntaxNodeAction`  will be called back for every single syntax node with matching kind in the tree.

I would propose registering a symbol action for analyzing all top level symbols (including parameters) and then an operation action with `OperationKind.VariableDeclarator` to analyze locals.

Reply:
So `RegisterSymbolAction` does "not find symbols for locals and parameters", but you propose registering a symbol action for "analyzing all top level symbols (**including parameters**)"?

Reply:
You can invoke [this](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Unfactored/Roslyn/Core/ISymbolExtensions.cs#L11) extension methods to get all parameters for member symbols.

Reply:
Cool thanks

Reply:
Hmm, what should I do to get type parameters e.g. `TFoo` in `Method<TFoo>(...)`? Apparently `RegisterSymbolAction` doesn't fire for `SymbolKind.TypeParameter` even though they have symbols...

Reply:
Nevermind; used `IMethodSymbol/INamedTypeSymbol.TypeParameters`